operations/rack_resiliency/manage_critical_services.md

```diff
@@ -0,0 +1,326 @@
+
+    ```bash
+    kubectl get ConfigMaps -n rack-resiliency
+    ```
+
+    Example output:
+
+    ```text
+    NAME                 DATA   AGE
+    istio-ca-root-cert   1      6d21h
+    kube-root-ca.crt     1      6d21h
+    rrs-mon-dynamic      2      3d4h
+    rrs-mon-static       11     3d4h
+    ```
+    
+**Verify that the list of critical services is present in the ConfigMap:**
+
+```bash
+kubectl describe configmap rrs-mon-static -n rack-resiliency
+```
+
+Example output:
+
+```text
+Name:         rrs-mon-static
+Namespace:    rack-resiliency
+Labels:       <none>
+Annotations:  <none>
+
+Data
+====
+ceph_pre_monitoring_delay:
+----
+60
+critical-service-config.json:
+----
+{
+  "critical_services": {
+    "cilium-operator": {
+      "namespace": "kube-system",
+      "type": "Deployment"
+    },
+    "coredns": {
+      "namespace": "kube-system",
+      "type": "Deployment"
+    },
+    "cray-power-control-bitnami-etcd": {
+      "namespace": "services",
+      "type": "StatefulSet"
+    },
+    "cray-powerdns-manager": {
+      "namespace": "services",
+      "type": "Deployment"
+    },
+    "cray-sls": {
+      "namespace": "services",
+      "type": "Deployment"
+    },
+    "cray-spire-postgres": {
+      "namespace": "spire",
+      "type": "StatefulSet"
+    },
+    ...
+}
+k8s_monitoring_total_time:
+----
+600
+k8s_pre_monitoring_delay:
+----
+40
+ceph_monitoring_polling_interval:
+----
+60
+default_message_level:
+----
+debug
+k8s_monitoring_polling_interval:
+----
+60
+last_updated_timestamp:
+----
+2025-06-30T04:49:35.298283Z
+log_dir:
+----
+/var/log/rr
+unit_of_time:
+----
+seconds
+ceph_monitoring_total_time:
+----
+600
+
+BinaryData
+====
+
+Events:  <none>
+```
+
+**In order to delete the service from the ConfigMap use the following command:**
+
+```bash
+kubectl edit configmap rrs-mon-static -n rack-resiliency
+```
+
+**Example output:**
+
+```yaml
+apiVersion: v1
+data:
+  ceph_monitoring_polling_interval: "60"
+  ceph_monitoring_total_time: "600"
+  ceph_pre_monitoring_delay: "60"
+  critical-service-config.json: |-
+    {
+      "critical_services": {
+        "cilium-operator": {
+          "namespace": "kube-system",
+          "type": "Deployment"
+        },
+        "coredns": {
+          "namespace": "kube-system",
+          "type": "Deployment"
+        },
+        "cray-activemq-artemis-operator-controller-manager": {
+          "namespace": "dvs",
+          "type": "Deployment"
+        },
+        "cray-capmc": {
+          "namespace": "services",
+          "type": "Deployment"
+        },
+        "cray-ceph-csi-cephfs-provisioner": {
+          "namespace": "ceph-cephfs",
+          "type": "Deployment"
+        },
+        "cray-ceph-csi-rbd-provisioner": {
+          "namespace": "ceph-rbd",
+          "type": "Deployment"
+        },
+    ...
+```
+
+**To remove a service:**
+1. Look for the services to be deleted inside the `critical-service-config.json` field, under the `data` section.
```

Comment:
These seem like they are steps to be followed after the previous steps. But the previous steps aren't marked as steps. This is confusing.